cypress/e2e/tests/pages/manager/cluster-provisioning-amazon-ec2-rke2.spec.ts

```diff
@@ -85,47 +87,48 @@ describe('Deploy RKE2 cluster using node driver on Amazon EC2', { testIsolation:
     createRKE2ClusterPage.nameNsDescription().name().set(this.rke2Ec2ClusterName);
     createRKE2ClusterPage.nameNsDescription().description().set(`${ this.rke2Ec2ClusterName }-description`);
 
-    // Get kubernetes versions - use an older version for initial creation, then upgrade later
-    cy.wait('@getRke2Releases').then(({ response }) => {
-      expect(response.statusCode).to.eq(200);
-      const index1 = response.body.data.length - 1;
-
-      // Store the latest version for upgrade test
-      k8sVersion = response.body.data[index1].id;
-      cy.wrap(k8sVersion).as('k8sVersion');
+    // Get kubernetes versions from the UI dropdown
+    // Index 0 is the "RKE2" header, so actual versions start at index 1
+    cy.wait('@getRke2Releases').its('response.statusCode').should('eq', 200);
+    createRKE2ClusterPage.basicsTab().kubernetesVersions().toggle();
+    createRKE2ClusterPage.basicsTab().kubernetesVersions().getOptions().then(($options) => {
+      // First RKE2 version (index 1) is the latest version
+      latestK8sVersion = Cypress.$($options[1]).text().trim();
+      cy.wrap(latestK8sVersion).as('latestK8sVersion');
+
+      // Second RKE2 version (index 2) is the older version for cluster creation
+      olderK8sVersion = Cypress.$($options[2]).text().trim();
```

Comment:
Could we add an `$options.length > 2` to prevent an `undefined` if there are less than three returned?